registry/storage/filewriter_test.go

```diff
@@ -165,8 +165,9 @@ func TestSimpleWrite(t *testing.T) {
 
 func TestBufferedFileWriter(t *testing.T) {
 	ctx := context.Background()
-	writer, err := newFileWriter(ctx, inmemory.New(), "/random")
 
+	writerp, err := newFileWriter(ctx, inmemory.New(), "/random")
+	writer := *writerp
```

Comment:
I'm sorry, but this is not a realistic way to use it. I am not sure what problem you are actually trying to solve.

If you could break this down into a use case that is not contrived, I might be able to understand what is going on here.